### PR TITLE
Move mongo-replicaset to stable

### PR DESCRIPTION
**Pull request by zonorti**
* Move to stable directory
* Update to mongo:3.4
* Replace PetSet with StatefulSet in docs
* Update readme examples and links
* Remove obsolete code from on-start.sh

### CODE REVIEW

**Comment by viglesiasce** [comment]
cc @foxish for review

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

**Comment by zonorti** [comment]
Looks like tests are still running for incubator, not stable.
Did I miss something?

**Comment by zonorti** [comment]
@viglesiasce I can see, that tests are running against /src/incubator/mongodb-replicaset, not /src/stable/mongodb-replicaset
Any hints on fixing that?

**Comment by viglesiasce** [comment]
Hey @iMelnik it might be because of the way the files were moved. Did you use `git mv` or `mv` and then `git add`?

**Comment by zonorti** [comment]
`mv` + `add`, I guess this was a wrong thing to do.
I'll fix conflicts and update the pr

**Comment by zonorti** [comment]
Had to remove readme.md changes to make tests pass. I guess I would have another PR for that

**Comment by foxish** [comment]
Thanks @iMelnik!

Sorry I dropped the ball on this one. 
I'll take a look today.

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ metadata:
     chart: "{{.Chart.Name}}-{{.Chart.Version}}"
     component: "{{.Release.Name}}-{{.Values.Component}}"
   annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
```
Comment (foxish) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: @iMelnik is it standard practice to include this annotation? I haven't seen it in the other examples.

**Comment by foxish** [comment]
LGTM except that one comment

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml (hunk 1 above)
Comment (prydonius) [reply stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: See #386, it was removed from other charts.

Comment (zonorti) [reply stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: my bad, merged it back from old branch

**Comment by zonorti** [comment]
@foxish removed the annotation

**Comment by foxish** [comment]
LGTM

**Comment by InAnimaTe** [comment]
Any updates on this? It looks mostly ready for primetime and I'd love to get using it ;)

**Comment by zonorti** [comment]
@InAnimaTe there are new conflicts, I would update the branch soon and, hopefully, @prydonius would accept it

**Comment by InAnimaTe** [comment]
Sweetness!

Hmm, also, just stumbled [upon](https://github.com/bitnami/bitnami-docker-mongodb#setting-up-a-replication) replication config for the bitnami image, which is used for the *stable/mongodb* chart. Wonder if that would make setting up replication any easier?

Additionally, I'm interested in helping contribute to either make this chart support or create a new chart which can spawn a [sharded cluster](https://docs.mongodb.com/manual/core/sharded-cluster-components/).

**Comment by zonorti** [comment]
@InAnimaTe with this chart applied multiple times I was able to create a sharded cluster with 3 replicasets and a replicaset for config servers. Yet there were some manual steps to connect replicasets to config servers.

**Comment by zonorti** [comment]
@prydonius @foxish @viglesiasce I fixed merge conflicts, I guess you could merge now.
Updates to readme would be in a separate PR

**Comment by InAnimaTe** [comment]
@iMelnik I'd love to hear more about you setting up a sharded cluster with this. We might be able to work together to eradicate those manual steps and make a simple toggle in `values` for people who might want a mongo cluster of that type.

Thanks for your (and others`) work on this!

**Comment by zonorti** [comment]
@InAnimaTe my workflow was:
Create 3 "shardsvr" replicasets, create "configsvr" replicset, add each replicaset to config via `sh.addShard( "<replSetName>/s1-mongo1.example.net:27017")`, add mongos instance connected to configsrv cluster.
Basically, I was following https://docs.mongodb.com/manual/tutorial/deploy-shard-cluster/
I have not tried creating variable number of replicasets in one chart, but it might be possible

**Comment by zonorti** [comment]
@prydonius @foxish @viglesiasce please have a look, AFAIK no more changes needed.

**Comment by lachie83** [comment]
@prydonius @foxish Is this GTG?

**Comment by foxish** [comment]
No more comments from my end, all good to go.

**Comment by seanknox** [comment]
Hi @iMelnik, can you bump the chart version to 1.0.0? I'l merge it then. Thanks!

**Comment by lachie83** [comment]
Removing LGTM until we come to a decision about this at #767

**Comment by lachie83** [comment]
@unguiculus @iMelnik how should we proceed given these two PRs?

**Comment by zonorti** [comment]
I would wait until #767 is merged and update this one

**Comment by lachie83** [comment]
@iMelnik will do

**Comment by zonorti** [comment]
@lachie83 rebased after #767, only minimal changes here.

**Comment by foxish** [comment]
@viglesiasce @lachie83 I think we can merge this.

**Comment by zonorti** [comment]
@viglesiasce @lachie83 @seanknox @prydonius any way to get this merged?

**Comment by lachie83** [comment]
I'll do another review.

**Comment by prydonius** [comment]
Looks fine to me, @lachie83 did you want to give this a quick look?

**Comment by lachie83** [comment]
LGTM

**Pull request merged**